src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +17,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<T> getAll(Class<? extends Machine> type);
```

Comment:
Bad bounds. 

You already have the T - type that parametrized Machines, so you can use it as bound. 

See example:
```
List<T> getAll(Class<? extends T> type);
```

Think why this is the better way